src/cheri_insts.sail

```diff
@@ -1007,7 +1007,7 @@ if not(cap_val.tag) then {
       let c = mem_read_cap(addr, aq, rl, false);
       match c {
         MemValue(v) => {
-          let cr = if   ptw_info == PTW_LOAD_CAP_ERR
+          let cr = if   ptw_info.ptw_lc_mod == PTW_LCM_CLR
```

Comment:
It'd be nice if the type by this point wasn't such that, were there to be a PTW_LCM_TRAP, it'd show up. Really we just want a bool (or a struct containing a bool, since ptw_info is shared among all paths), so this just becomes `if ptw_info.lcm_clear`. Unfortunately since it's ext_ptw all the way through, `ext_get_ptw_error` needs enough info to know whether it was a CHERI page fault or not so that `translateException` knows which type of page fault to raise, which means that ptw_info has to have a bit of extra internal state, though that can be just another bool for whether to give `PTW_No_Permission` or `PTW_Ext_Error`.

Reply:
Unfortunately, the `PTW_LCM_TRAP` *has* to show up here: it's tag-dependent and the PTW has no access to the data path.

Reply:
I didn't mean that the signal has to be here. I meant that it was here _in this form_ with potentially states we don't care about (ie, in the full LCM world, you have PTW_LCM_TRAP, PTW_LCM_CLR and PTW_LCM_OK, and all three are possible values according to the type here, but PTW_LCM_TRAP is known to never happen; really we just want a single bool by this point saying whether to clear or preserve the tag, as the trapping has already happened). But as I said above, that requires refactoring the sail-riscv ext_ptw code to split things out into different types at different stages, so this is fine for now.